src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -36,6 +38,21 @@ public ODataSerializerContext()
         /// </param>
         /// <remarks>This constructor is used to construct the serializer context for writing nested and expanded properties.</remarks>
         public ODataSerializerContext(ResourceContext resource, SelectExpandClause selectExpandClause, IEdmProperty edmProperty)
+            : this(resource, selectExpandClause, edmProperty, null, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ODataSerializerContext"/> class.
+        /// </summary>
+        /// <param name="resource">The resource whose navigation property is being expanded.</param>
+        /// <param name="selectExpandClause">The <see cref="SelectExpandClause"/> for the navigation property being expanded.</param>
+        /// <param name="edmProperty">The complex property being nested or the navigation property being expanded.
+        /// If the resource property is the dynamic complex, the resource property is null.
+        /// </param>
+        /// <param name="queryOptions">The <see cref="ODataQueryOptions"/> for the navigation property being expanded.</param>
+        /// <param name="expandedItem">The <see cref="ExpandedReferenceSelectItem"/> for the navigation property being expanded.></param>
+        public ODataSerializerContext(ResourceContext resource, SelectExpandClause selectExpandClause, IEdmProperty edmProperty, ODataQueryOptions queryOptions, ExpandedReferenceSelectItem expandedItem)
```

Comment:
>ODataSerializerContext [](start = 15, length = 22)

I would rather not have a (public) constructor that takes both selectExpandClause and queryOptions. Ideally we would just take queryOptions, but I understand for legacy reasons we still have to support the public constructor that just takes a selectExpandClause.

Instead of having selectExpandClause in the new (public) overload, can we just take queryOptions, and have both the old and new constructor call into a private method to set the common properties?